### DISCUSSION TITLE
A question about simulating a contracting ring

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#21672] [Q&A Modules: Solid mechanics] [answered]
Hi all,

I want to simulate a contracting ring by increasing its inner radius but not changing its outer radius, just like what the figure below shows. Directly setting the displacement BC at the inner bound is not OK because I don't want this contract to cause extra stress respond. Does any one know how to realize this? One of my thoughts is to derive the after-contract position (xe, ye) of any point (x, y) to get the displacement field u(x, y) = (xe, ye) - (x, y). Then we can get the strain tensor εij = 0.5 * (∂ui/∂xj + ∂uj/∂xi) to construct an eigen strain material to realize the contract. Is this approach correct?

![contracting_ring](https://user-images.githubusercontent.com/34068460/180637580-be5c86bf-d3c3-4ac8-b87d-1e324121e0c7.JPG)

**Comment by WilkAndy** [comment, marked as answer]
Firstly, whatever you do, make sure you solve this using RZ coordinates in MOOSE.

Secondly, i don't understand how you can deform the annulus and maintain stress=0.  The only stress=0 solution is when the displacement is zero (except for rigid-body movement).  Generally, when displacement!=0 then stress!=0.  i think i'm fundamentally misunderstanding something here.  Can you explain further?  Eg, do you also have temperature?

**Reply by xuxiaobei1995** [reply]
My understanding is that if we can give the ring an eigen strain that make it contract "uniformly", it may maintain stress = 0, just like a uniform thermal expansion process.  More specifically, if we can guarantee that the relative position of each point in the ring is not changed during the contract process (r1/r2 = r3/r4 in the figure), the stress may not be generated. Is this a wrong idea?

![uniform](https://user-images.githubusercontent.com/34068460/180645927-c9b18028-f584-4f6a-8248-c46027907ab1.JPG)

**Reply by WilkAndy** [reply]
I think i don't have enough imagination to understand at the moment.  Where does this eigenstrain come from?  In your thermal expansion example, the eigenstrain comes from a temperature change, so while the stress=0, some other property (the temperature) has changed, which causes the displacement.  So, in your example, what's the other property that's going to change and cause displacement != 0.  Clearly, the displacement will depend on that property.  (For instance, if it were temperature and your annulus had anisotropic or heterogeneous thermal-expansion properties, then the displacement could be quite complicated compared with the isotropic, homogeneous case.)

**Reply by xuxiaobei1995** [reply]
Maybe what I said before is ambiguous ... In fact, my aim is just to find a way to change the geometry condition of the domain with time. More specifically, the user can specify the inner radius varying with time Rin(t) and the geometry will be changed with time in the simulation depending on Rin(t) while the outer radius is not changed. Since it is a pure geometry effect, I don't want this change to generate stress respond.

I think a possible way to realize this function is to create a NEW eigen strain for the material so that the annulus will deform as what Rin(t) describes with no stress respond. But how the annulus deforms with no stress respond?  my thought is that if the deformation is "uniform", the extra stress may not be generated, just like the uniform thermal expansion process. the "uniform" here means the relative position of each point in the ring is not changed during the contract process (r1/r2 = r3/r4 in the figure). So based on this geometry relation, we can get the displacement field u(x,y,Rin(t)), then we can get the eigen strain tensor εij = 0.5 * (∂ui/∂xj + ∂uj/∂xi), based on which the new eigen strain object (e.g., ComputeContractEigenStrain) will be developed. The inputs of the eigen strain object are Rin(t) and Rout. Do you think this approach is reasonable?

**Reply by WilkAndy** [reply]
I might be understanding your question.  Is this true: you know that in real life, the deformation you describe will result in stress!=0, but you want to trick MOOSE so that it thinks stress=0 ?   If so, then i think just create your EigenStrain as you describe above.

**Reply by xuxiaobei1995** [reply]
That is exactly what I meant. Thank you!
